stm32-modules/include/heater-shaker/heater-shaker/system_task.hpp

```diff
@@ -217,83 +229,142 @@ class SystemTask {
+        } else {
+            _led_state.current_color = msg.color;
+            _led_state.current_mode = msg.mode;
+        }
+        if ((_led_state.current_color == LED_COLOR::RED_WHITE) ||
+            (_led_state.current_color == LED_COLOR::RED_AMBER) ||
+            (_led_state.current_color == LED_COLOR::WHITE_AMBER)) {
+            _led_state.led_alternate_colors = true;
+            if (_led_state.current_color == LED_COLOR::RED_WHITE) {
+                _led_state.led_color_1 = LED_COLOR::RED;
+                _led_state.led_color_2 = LED_COLOR::WHITE;
+            } else if (_led_state.current_color == LED_COLOR::RED_AMBER) {
+                _led_state.led_color_1 = LED_COLOR::RED;
+                _led_state.led_color_2 = LED_COLOR::AMBER;
+            } else if (_led_state.current_color == LED_COLOR::WHITE_AMBER) {
+                _led_state.led_color_1 = LED_COLOR::WHITE;
+                _led_state.led_color_2 = LED_COLOR::AMBER;
             }
-            if (bStatus) {
-                if (led_blink_state.status == LEDBlinkState::BLINK_ON_WAITING) {
-                    if (policy.start_set_led(LED_MODE::WHITE_OFF) !=
-                        errors::ErrorCode::NO_ERROR) {
-                        bStatus = false;
-                        static_cast<void>(
-                            task_registry->comms->get_message_queue().try_send(
-                                messages::ErrorMessage{
-                                    .code = errors::ErrorCode::
-                                        SYSTEM_LED_TRANSMIT_ERROR}));
-                    }
-                    if (bStatus) {
-                        led_blink_state.status =
-                            LEDBlinkState::BLINK_OFF_WAITING;
-                    }
-                } else if (led_blink_state.status ==
-                           LEDBlinkState::BLINK_OFF_WAITING) {
-                    if (policy.start_set_led(LED_MODE::WHITE_ON) !=
-                        errors::ErrorCode::NO_ERROR) {
-                        bStatus = false;
-                        static_cast<void>(
-                            task_registry->comms->get_message_queue().try_send(
-                                messages::ErrorMessage{
-                                    .code = errors::ErrorCode::
-                                        SYSTEM_LED_TRANSMIT_ERROR}));
-                    }
-                    if (bStatus) {
-                        led_blink_state.status =
-                            LEDBlinkState::BLINK_ON_WAITING;
+        } else {
+            _led_state.led_alternate_colors = false;
+        }
+    }
+
+    template <typename Policy>
+    auto visit_message(const messages::UpdateLEDMessage& msg, Policy& policy)
+        -> void {
+        bool bStatus = true;
+        if (!policy.check_I2C_ready()) {
+            bStatus = false;
+            static_cast<void>(
+                task_registry->comms->get_message_queue().try_send(
```

Comment:
I'm not sure that the firmware needs to send error messages to the host about LED I2C errors. If this _is_ what we want, maybe the error state should latch so the task isn't sending the same error every 25ms.